### ISSUE TITLE
Multiple #@local lines?

### ISSUE THREAD

**Issue by cdwensley** [gap-system/gap#5050]
Having just discovered that `#@local` lines are recommended for tests I decided to adjust the tests in Utils. 
The Test documentation states "Multiple #@local lines may be used." but I have found this not to be the case. 
The example below is, of course, trivial, but groups.tst requires 4 lines of variables, producing a string of length 304. 
Is there an upper bound one should be concerned about? 

### Observed behaviour
Adding the following lines at the start of number.tst: 
```
#@local  n, p; 
#@local  sum, iter; 
```
produces the following error when the test is run (in both 4r12 and dev): 
```
gap> Read("tst/number.g"); 
running number.tst
Syntax error: while parsing a function: statement or 'end' expected in stream:\
1
_EVALSTRINGTMP:=(function() local n, p; , sum, iter;;return GetCurrentLVars();\
 end)();
                                        ^
Error, Could not evaluate string.
```

**Comment by cdwensley** [comment]
Furthermore, the documentation should make it clear that if a #@local list is given, then _every_ variable in the test must be included.  Changing number.tst to start with  "#@local  n, p, sum, iter;"  still produces an error because the variable i has not been included: 
########> Diff in tst/number.tst:56 
# Input is:
for i in [1..100] do  p := NextIterator(iter);  od;
# Expected output:
# But found:
Syntax warning: Unbound global variable in stream:1
for i in [1..100] do  p := NextIterator(iter);  od;
    ^
########

**Comment by fingolfin** [comment]
Internally what happens is that when `@local` is used then the code in the .tst file is executed in the context of a GAP function with an equivalent `local` statement. Since functions only support a single `local` statement, this then extends to `@local` statements... But I guess we could make it work by collecting all such statements and concatenating them?

**Comment by fingolfin** [comment]
Actually, we already do that. The problem here is that you put a trailing semicolon in. Remove those, and it should work.

What we then perhaps should improve is validation of the values given after `@local`: it should be a comma separated list of valid GAP identifiers. No semicolons, no parentheses etc.

**Comment by fingolfin** [comment]
It would probably be a good idea to add concrete examples to the `Test` function manual entry that showcase `@local` but also `@if` etc.

**Comment by cdwensley** [comment]
Now understood, so have rewritten the description.  Have altered utils/tst/number.tst to provide an example.  The @if construction does not appear to be much used, but I have added an example from recog which seems to work.  I expect others will suggest a better example?

**Comment by fingolfin** [comment]
Resolved by #5051
